From 7cf0e8608bda5bbb0e9ad9930c18cf6d8bd11be1 Mon Sep 17 00:00:00 2001
From: Kiwi Browser <40272275+kiwibrowser@users.noreply.github.com>
Date: Sun, 13 Aug 2023 12:07:38 +0300
Subject: [PATCH 492/506] Download artifacts before creating the release so
 there is less waiting time before the assets can appear on a release

---
 .../workflows/build_and_sign_release_apk.yml  | 43 ++++++++++---------
 1 file changed, 22 insertions(+), 21 deletions(-)

diff --git a/.github/workflows/build_and_sign_release_apk.yml b/.github/workflows/build_and_sign_release_apk.yml
index a8707e96..23c6bc28 100644
--- a/.github/workflows/build_and_sign_release_apk.yml
+++ b/.github/workflows/build_and_sign_release_apk.yml
@@ -10,7 +10,7 @@ on:
         required: true
         default: 'no'
       announceOnMessengers:
-        description: 'Announce new release on Discord and Telegram channel'     
+        description: 'Announce new release on Discord and Telegram channels'     
         required: true
         default: 'yes'
 
@@ -160,6 +160,27 @@ jobs:
           echo 'user = storage' >> $HOME/.config/rclone/rclone.conf
           echo 'key_pem = ${{ secrets.STORAGE_KEY }}' >> $HOME/.config/rclone/rclone.conf
 
+
+      - name: Downloading artifact (arm)
+        uses: actions/download-artifact@v1
+        with:
+          name: apk-arm
+
+      - name: Downloading artifact (arm64)
+        uses: actions/download-artifact@v1
+        with:
+          name: apk-arm64
+
+      - name: Downloading artifact (x86)
+        uses: actions/download-artifact@v1
+        with:
+          name: apk-x86
+
+      - name: Downloading artifact (x64)
+        uses: actions/download-artifact@v1
+        with:
+          name: apk-x64
+
       - name: Read last commits
         id: read_commits
         run: |
@@ -229,26 +250,6 @@ jobs:
                 
             Essentially, if you downloaded Kiwi from an app store, you need to use the "-playstore.apk" files or uninstall the version of Kiwi you have and start using the "-github.apk" version.
 
-      - name: Downloading artifact (arm)
-        uses: actions/download-artifact@v1
-        with:
-          name: apk-arm
-
-      - name: Downloading artifact (arm64)
-        uses: actions/download-artifact@v1
-        with:
-          name: apk-arm64
-
-      - name: Downloading artifact (x86)
-        uses: actions/download-artifact@v1
-        with:
-          name: apk-x86
-
-      - name: Downloading artifact (x64)
-        uses: actions/download-artifact@v1
-        with:
-          name: apk-x64
-
       - name: Signing and uploading release to GitHub and send to the Play Store
         if: ${{ github.repository_owner == 'kiwibrowser' && contains(github.event.inputs.prepareRelease, 'yes') }}
         run: |
